core/internal/api/api.go

```diff
@@ -181,6 +181,9 @@ type ClientOptions struct {
 	// on the request and response. Need to make sure that the response body is
 	// available to read by later stages.
 	NetworkPeeker Peeker
+
+	// Proxy function to use for requests.
```

Comment:
needs more info here :)

* what's the purpose of this?
* what are the inputs?
* what are the expected outputs?